spring-vault-core/src/main/java/org/springframework/vault/core/lease/SecretLeaseContainer.java

```diff
@@ -375,7 +385,7 @@ public void destroy() throws Exception {
 					Lease lease = entry.getValue().getLease();
 					entry.getValue().disableScheduleRenewal();
 
-					if (lease != null) {
+					if (lease != null && !lease.isRotatingGenericLease()) {
```

Comment:
There's a bit more logic to `isRotatingGenericLease`:

1. empty/null lease ID
1. non-renewable
1. positive lease duration

The first two give you some pretty good confidence you're looking at a generic secret. The last one tells you whether or not you should expect it to rotate.